src/citrine/_session.py

```diff
@@ -128,6 +128,28 @@ def delete_resource(self, path: str) -> dict:
         """DELETE a particular resource as JSON."""
         return self.checked_delete(path).json()
 
+    @staticmethod
+    def cursor_paged_resource(base_method: Callable[..., dict], path: str, *args,
+                              forward: bool = True, per_page: int = 100, **kwargs):
+        """
+        Returns a flat generator of results for an API query.
+
+        Results are fetched in chunks of size `per_page` and loaded lazily.
+        """
+        params = kwargs.get('params', {})
+        params['forward'] = forward
+        params['ascending'] = forward
+        params['per_page'] = per_page
+        kwargs['params'] = params
+        while True:
+            response_json = base_method(path, *args, **kwargs)
+            for obj in response_json['contents']:
+                yield obj
+            cursor = response_json.get('next')
```

Comment:
What will `cursor` be in general? Is it always an integer?

Reply:
It's actually an arbitrary string which the client doesn't need to understand. Integers are just easy to use in the test. There are a lot of pieces here, which I will link, but you may be better off just reading about general cursor-base paging on the internet. I can find you a good article if you want.

Reply:
I've tried to made some decent docs here. If you're up for reading some scaladocs and source, here are some relevant classes to look at:

General interface for query params:
https://github.com/CitrineInformatics/platform-backend/blob/develop/code/rest/akka-http/src/main/scala/io/citrine/rest/akka/http/modules/PagingSupport.scala
The resulting information about the page to fetch:
https://github.com/CitrineInformatics/platform-backend/blob/develop/code/rest/rest-base/src/main/scala/io/citrine/rest/model/Page.scala
The cursor, itself (not much here):
https://github.com/CitrineInformatics/platform-backend/blob/develop/code/rest/rest-base/src/main/scala/io/citrine/rest/model/Cursor.scala
Example of interpreting a cursor:
https://github.com/CitrineInformatics/platform-backend/blob/develop/code/faraday/dc-dynamo/src/main/scala/io/citrine/faraday/dynamo/DynamoCursor.scala
The results returned:
https://github.com/CitrineInformatics/platform-backend/blob/develop/code/rest/rest-base/src/main/scala/io/citrine/rest/model/PagedResult.scala
Bonus points for reading the code that actually uses the dynamo cursor:
https://github.com/CitrineInformatics/platform-backend/blob/develop/code/faraday/dc-dynamo/src/main/scala/io/citrine/faraday/dynamo/service/DynamoReadDataService.scala#L470

Reply:
An arbitrary string seems fine in this case; it doesn't need to be human-readable.

Thanks for the links. I've started looking at them for my own edification, but I don't think it's necessary for me to review this PR. I was just trying to get a sense of what the client needs to know. I don't think the other method you mentioned (backend returns prev/next links) is conceptually much simpler. Instead of extracting the `cursor` from `response_json` and feeding that into the `params` of the subsequent call, you'd instead have to extract the link to prev/next from (presumably) `response_json` and feed that into the subsequent call somehow. The actual code might be slightly shorter, but I think that the mental energy from deciding "I'd like to implement pagination in this client" to having it functioning would be more or less equal.

I might be misunderstanding how the alternative pagination method would work, but for now I see no reason not to implement it this way.